File: python/baseball-game.py
class Solution:
    def calPoints(self, operations: List[str]) -> int:
        records = []
        for operation in operations:
            if operation == "+":
                records.append(records[-1] + records[-2])
            elif operation == "D":
                records.append(records[-1] * 2)
            elif operation == "C":
                records.pop()
            else:
                records.append(int(operation))
        return sum(records)

File: rust/baseball-game.rs
impl Solution {
    pub fn cal_points(operations: Vec<String>) -> i32 {
        let mut records = Vec::new();
        for operation in operations {
            if operation == "C" {
                records.pop();
            } else if operation == "D" {
                records.push(records[records.len() - 1] * 2);
            } else if operation == "+" {
                records.push(records[records.len() - 1] + records[records.len() - 2]);
            } else {
                records.push(operation.parse().unwrap());
            }
        }
        return records.iter().sum();
    }
}
